### DISCUSSION TITLE
Problema com publish no GitHub

### DISCUSSION THREAD

**Discussion by JaqueManzano** [balta-io/forum#461] [Tira Dúvidas] [answered]
Boa noite Balta!

Eu de novo com problema de publicação rsrs 
O meu yml é este, 

![image](https://github.com/balta-io/forum/assets/143463393/7086f35b-aba1-4a04-90ce-9b1c337152ba)

e estou recebendo a seguinte mensagem de erro: 

![image](https://github.com/balta-io/forum/assets/143463393/b8f26d83-4692-45f1-9655-c466779521c3)

Vi em algumas discussões pessoas dizendo que o problema ocorre por conta de estarmos tentando usar o GITHUB_TOKEN em uma conta GitHub pessoal e não em uma conta organizacional, pode ser este o problema?

O link utilizado no yml está correto e consigo acessar normalmente através do navegador.

![image](https://github.com/balta-io/forum/assets/143463393/82fa28fe-5055-4ee6-bd5c-c1de88a0ec01)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @JaqueManzano, tudo bem por ai?

Normalmente este problema ocorre por que sua conta pessoal (Onde o PAT - Public Access Token é gerado) não tem acesso ou permissão para consumir o pacote publicado no GitHub Packages da organização.

Se você gerou um PAT e só depois deu permissão para sua conta na empresa, é preciso gerar um novo PAT.
Então na dúvida, tenta o seguinte:
1 - Garante que seu perfil tem acesso ao repositório e pacotes da organização no GitHub
2 - Gera um novo PAT e tenta novamente.

Testa e me fala o que aconteceu.

**Reply by JaqueManzano** [reply]
Tudo certo!

Então @andrebaltieri , ainda não cheguei na parte do PAT, que seria a configuração do Nuget.config para a utilização do pacote publicado.
Estou tentando fazer a publicação do primeiro projeto ( MyCustomPrivatePackage ) no Nuget do Github utilizando apenas o GITHUB_TOKEN... Até este ponto do curso não é citado a geração do PAT.

**Reply by andrebaltieri** [reply]
Bom dia, @JaqueManzano, como vai?
Me desculpa, confundi então!

Sobre o Token para publicação, ele vai validar 3 coisas:
- O nome do seu pacote no Nuget deve ser o mesmo nome do repositório
- O Token precisa de acesso ao repositório
- No seu CSPROJ, a configuração tem que apontar para o repositório

Se tudo isto estiver OK, tenta gerar um novo Token e fazer um novo teste por favor.

**Reply by JaqueManzano** [reply]
O GITHUB_TOKEN é o Token que você cita que não precisa ser criado @andrebaltieri , é o Token gerenciado pelo próprio GitHub.

**Reply by andrebaltieri** [reply]
@JaqueManzano entendi, desculpa novamente... que estranho... você tem acesso para criar pacotes nesta organização de destino?

Consegue conferir por favor, aqui tem todas as permissões necessárias:
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages

No vídeo, eu publiquei na organização do balta, no qual sou proprietário... O erro 403 que você recebe é justamente por que não tem permissão para publicação do pacote.

**Reply by JaqueManzano** [reply]
Boa noite!

Realmente era questão de permissão do GITHUB_TOKEN na minha conta pessoal do Github.
Com base na documentação que você me passou, na sessão "Configuring the default GITHUB_TOKEN permissions" consegui encontrar como alterar a permissão do meu token em "Workflow Permissions" fazendo com que meu pacote fosse publicado corretamente.

Acessei o meu repositório >> Settings  >> Na aba "Actions" acessei "General"
Em "Workflow permissions" marquei a opção "Read and write permissions"

Obrigada!

**Reply by andrebaltieri** [reply]
Show @JaqueManzano, muito obrigado por compartilhar a solução!
